astropy/units/typing.py

```diff
@@ -0,0 +1,66 @@
+"""Typing module for supporting type annotations related to :mod:`~astropy.units`."""
+
+from __future__ import annotations
+
+__all__ = ["QuantityLike"]
+
+
+from typing import TYPE_CHECKING, Union
+
+import numpy.typing as npt
+
+from astropy.units import Quantity
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+# Note: Quantity is technically covered by npt.ArrayLike, but we want to
+# explicitly include it here so that it is clear that we are also including
+# Quantity objects in the definition of QuantityLike.
+QuantityLike: TypeAlias = Union[Quantity, npt.ArrayLike]
+"""Type alias for a quantity-like object.
```

Comment:
```suggestion
"""Type alias for a :term:`quantity-like` object.
```
Should we link to the glossary term?

Reply:
Can do. I worry about circularity since I pointed the glossary term here. I think this code definition should be definitive and the glossary is a convenient centralization of important terms.

Reply:
@mhvk, what do you think? If this object is the actual definition and the glossary now points here, should we backlink?

Reply:
I think there is indeed no need to back link. I also prefer to at least have the first line of the docstring just be plain text, without mark-up (though this docstring will not easily get seen from `ipython`).